### PR TITLE
Feature/add evolve data migration

### PR DESCRIPTION
**Pull request by Jamie-Unosquare**
Added migrations and interface settings.
Resolved conflicts with Swagger change

Logger to be added in future

### CODE REVIEW

File: v2/AdminCore.DAL/Entity Framework/EntityFrameworkContext.cs [hunk 1]
```diff
@@ -45,6 +45,11 @@ public EntityFrameworkContext(AdminCoreContext adminCoreContext)
     /// </summary>
     public IRepository<Employee> EmployeeRepository => _EmployeeRepository ?? (_EmployeeRepository = new EntityFrameworkRepository<Employee>(this));
 
+    public IDbConnection RetrieveConnection()
```
Comment (BrownMark) [inline v2/AdminCore.DAL/Entity Framework/EntityFrameworkContext.cs]: Don't need this anymore.

File: v2/AdminCore.Common/Interfaces/IOutput.cs [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+//   Admincore
+// </copyright>
+// <summary>
+//   Defines the IDataMigration type.
+// </summary>
+// --------------------------------------------------------------------------------------------------------------------
+
+namespace AdminCore.Common.Interfaces
+{
+    public interface IOutput
```
Comment (BrownMark) [inline v2/AdminCore.Common/Interfaces/IOutput.cs]: Let's delete this, was necessary for a console app unit tests.

File: v2/AdminCore.DAL/IDatabaseContext.cs [hunk 3]
```diff
@@ -25,5 +26,7 @@ public interface IDatabaseContext
     /// The save changes.
     /// </summary>
     void SaveChanges();
+
+    IDbConnection RetrieveConnection();
```
Comment (BrownMark) [inline v2/AdminCore.DAL/IDatabaseContext.cs]: Please delete.

File: v2/AdminCore.Services/Configuration/EvolveDataMigration.cs [hunk 4]
```diff
@@ -0,0 +1,89 @@
...
+    }
+
+    /// <summary>
+    /// The retrieve database connection.
+    /// </summary>
+    /// <returns>
+    /// The <see cref="IDbConnection"/>.
+    /// </returns>
+    public virtual IDbConnection RetrieveDatabaseConnection()
+    {
```
Comment (BrownMark) [inline v2/AdminCore.Services/Configuration/EvolveDataMigration.cs:63]: Let's pull the connectionstring from the common IConfiguration implementation

**Review by BrownMark — CHANGES_REQUESTED** [review]

**Review by eoinmcafee — APPROVED** [review]

**Review by BrownMark — APPROVED** [review]

**Pull request merged**